### DISCUSSION TITLE
Does ConnectorX work for MS SQL Server with trusted connection on Linux?

### DISCUSSION THREAD

**Discussion by athensofamerica** [sfu-db/connector-x#684] [Q&A]
Has any body know how to get ConnectorX working with MS SQL server with trusted connection on Linux? What driver should we use? and What connection string look like. I am keep getting Time out in bb8 RuntimeError. Here is the connection string that I used:
f"mssql://{server}/{database}?driver=ODBC+Driver+17+for+SQL+Server&trusted_connection=yes"

Replace {server} and {database} with actual database server and database? 
I also tried with driver=SQL+Server and that did help either. But it also got error messages before final timeout error:
Login failed for user '', code=18456

Any helps will be greatly appreciated.

Update:
After checking with the source code, looks like the trusted_connection parameter should be set to "true" rather than "yes". However, this doesn't help either. I tried on Windows and it works fine. So, this appears to be an Linux only issue. In any case, using SQLAlchemy  + pyodbc and trusted_connection=yes works fine. In other words, Linux seems do have support for integrated authentication and wonder if the MSSQL datasource could be enhanced to support Linux as well.

**Comment by strefli3** [comment]
I have a similar problem on my end authenticating with

```
f"mssql://{SQL_READ_USERNAME}:{SQL_READ_PASSWORD}@{SQL_SERVER}:1433/{DB_NAME}?trust_server_certificate=true&encrypt=true&trusted_connection=false"
```

which works fine in Windows, but not on Linux which throws a `RuntimeError: Timed out in bb8`. Not that this is not a `trusted_connection`.

I have tried with both Python 3.9 and 3.11  using `connectorx` 0.3.3. I also tries 0.4.0 on Python 3.11. Both pyodbc and sqlalchemy work in  both Windows and Linux.
